typings/styled-components.d.ts

```diff
@@ -199,6 +199,8 @@ export interface ThemedStyledComponentsModule<T> {
 declare const styled: StyledInterface;
 
 export const css: ThemedCssFunction<any>;
+export function withTheme<C extends Component<any>>(component: C): C;
```

Comment:
An inner class is required to have `theme` prop and an implementer may want to have it required:
```ts
interface MyComponentProps {
  theme: MyTheme; // I want have this required
  text: string;
}
class MyComponent extends React.Component<MyComponentProps, {}> {
  ...
}

<MyComponentProps theme={theme} text={text} /> // ok
<MyComponentProps text={text} /> // error, I can't use without theme
```

so after applying the `withTheme`, the component will lose `theme` prop:

```ts
interface MyThemedComponentProps {
  text: string;
}
const MyThemedComponent: ComponentClass<MyThemedComponentProps> = withTheme(MyComponent);

<MyThemedComponent text={text} /> // now that is ok
<MyThemedComponent theme={theme} text={text} /> // and now that is an error, because there's no theme prop anymore
```